File: pytools/data/data.code.lua.server/ReqTestSend.lua
module("ReqTestSend", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqTestSend})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.id_u8 = 0
	self.role_base = nil
	self.id_f32 = {}
	self.id_op_u8_flag = 0
	self.id_op_u8 = 0
	self.op_role_base_flag = 0
	self.op_role_base = nil
end


function decode(self, pack)
	self.id_u8 = pack:readU8()
	self.role_base = MsgRoleBase.new():decode(pack)
	id_f32_count = pack:readU16()
	for i = 1, id_f32_count do
		table.insert(self.id_f32, pack:readF32())
	end
	self.id_op_u8_flag = pack:readU8()
	if self.id_op_u8_flag then
		self.id_op_u8 = pack:readU8()
	end
	self.op_role_base_flag = pack:readU8()
	if self.op_role_base_flag then
		self.op_role_base = MsgRoleBase.new():decode(pack)
	end
	return self
end


function getIdU8(self)
	return self.id_u8
end

function getRoleBase(self)
	return self.role_base
end

function getIdF32(self)
	return self.id_f32
end

function getIdOpU8(self)
	return self.id_op_u8
end

function getOpRoleBase(self)
	return self.op_role_base
end

File: pytools/data/data.code.php.server/ReqTestSend.php
<?php
namespace protocol;

use network\Packet;


class ReqTestSend
{
	private $idU8 = 0;
	private $roleBase = null;
	private $idF32 = [];
	private $idOpU8Flag = 0;
	private $idOpU8 = 0;
	private $opRoleBaseFlag = 0;
	private $opRoleBase = null;


	public function __construct(&$packet)
	{
		$this->decode($packet);
	}

	public function decode(&$packet)
	{
		$this->idU8 = $packet->readU8();
		$this->roleBase = new MsgRoleBase($packet);
		$idF32Count = $packet->readU16();
		for ($i = 0; $i < $idF32Count; $i++)
		{
			array_push($this->idF32, $packet->readF32());
		}
		$this->idOpU8Flag = $packet->readU8();
		if ($this->idOpU8Flag == 1)
		{
			$this->idOpU8 = $packet->readU8();
		}
		$this->opRoleBaseFlag = $packet->readU8();
		if ($this->opRoleBaseFlag == 1)
		{
			$this->opRoleBase = new MsgRoleBase($packet);
		}
	}


	public function getIdU8()
	{
		return $this->idU8;
	}

	public function getRoleBase()
	{
		return $this->roleBase;
	}

	public function getIdF32()
	{
		return $this->idF32;
	}

	public function getIdOpU8()
	{
		return $this->idOpU8;
	}

	public function getOpRoleBase()
	{
		return $this->opRoleBase;
	}

}
